auth0/src/main/java/com/auth0/android/authentication/storage/CryptoUtil.java

```diff
@@ -162,57 +207,111 @@ public CryptoUtil(@NonNull Context context, @NonNull Storage storage, @NonNull S
         return new KeyStore.PrivateKeyEntry(privateKey, new Certificate[]{certificate});
     }
 
-    //Used to delete recreate the key pair in case of error
+    /**
+     * Removes the AES and RSA keys generated in a previous execution.
+     * Used when we want the next call to {@link #encrypt(byte[])} or {@link #decrypt(byte[])}
+     * to recreate the keys.
+     */
     private void deleteKeys() {
+        storage.remove(KEY_ALIAS);
+        storage.remove(KEY_IV_ALIAS);
```

Comment:
moved up to ensure this is called regardless the keystore calls succeed or not.

Reply:
Separated this into 2 separate calls, since some scenarios required only the AES keys to be recreated